src/query/service/src/pipelines/processors/transforms/hash_join/hash_join_build_state.rs

```diff
@@ -227,8 +254,14 @@ impl HashJoinBuildState {
             if self.hash_join_state.need_mark_scan() {
                 build_state.mark_scan_map.push(block_mark_scan_map);
             }
+
             build_state.generation_state.build_num_rows += data_block.num_rows();
             build_state.generation_state.chunks.push(data_block);
+
+            if self.hash_join_state.merge_into_target_table_index != DUMMY_TABLE_INDEX {
```

Comment:
`self.hash_join_state.merge_into_target_table_index != DUMMY_TABLE_INDEX` is called multiple times, better put into a new function named `is_merge_to` ?